src/middleware/ember_html.rs

```diff
@@ -26,11 +67,63 @@ pub async fn serve_html(request: Request, next: Next) -> Response {
         .iter()
         .any(|val| val.to_str().unwrap_or_default().contains("html"))
     {
+        if !matches!(*request.method(), Method::HEAD | Method::GET) {
+            let headers =
+                HeaderMap::from_iter([(header::ALLOW, HeaderValue::from_static("GET,HEAD"))]);
+            return (StatusCode::METHOD_NOT_ALLOWED, headers).into_response();
+        }
+
+        // Come up with an Open Graph image URL. In case a crate page is requested,
+        // we use the crate's name and the OG image base URL from config to
+        // generate one, otherwise we use the fallback image.
+        let og_image_url = 'og: {
+            if let Some(suffix) = path.strip_prefix(PATH_PREFIX_CRATES) {
+                let len = suffix.find('/').unwrap_or(suffix.len());
+                let krate = &suffix[..len];
```

Comment:
might make sense to extract a `extract_crate_name(...)` fn. that way it's a easier to write a few tests for it too.